File: Plugins/Aspose_Tasks_Java_for_Ruby/lib/asposetasksjava/Calendars/createcalendar.rb
module Asposetasksjava
  module CreateCalendar
    def initialize()
        data_dir = File.dirname(File.dirname(File.dirname(File.dirname(__FILE__)))) + '/data/'
                
        # Instantiate project object
        project = Rjb::import('com.aspose.tasks.Project').new(data_dir + 'test_tasks.mpp')

        # Define Calendar
    	cal1 = project.getCalendars().add("no info")
    	cal2 = project.getCalendars().add("no name")
    	cal3 = project.getCalendars().add("cal3")

    	# Save the Project
    	project.save(data_dir + "CreateCalendar.xml", Rjb::import('com.aspose.tasks.SaveFileFormat').XML)

    	puts "Created calendar, please check the output file."
    end
  end
end

File: Plugins/Aspose_Tasks_Java_for_PHP/src/aspose/tasks/WorkingWithCalendars/CreateCalendar.php
<?php
namespace Aspose\Tasks\WorkingWithCalendars;

use com\aspose\tasks\Project as Project;
use com\aspose\tasks\SaveFileFormat as SaveFileFormat;

class CreateCalendar{

    public static function run($dataDir=null){

        # Instantiate project object
        $project = new Project($dataDir . 'test_tasks.mpp');

        # Define Calendar
    	$cal1 = $project->getCalendars()->add("no info");
    	$cal2 = $project->getCalendars()->add("no name");
    	$cal3 = $project->getCalendars()->add("cal3");

    	# Save the Project
        $saveFileFormat=new SaveFileFormat();
    	$project->save($dataDir . "CreateCalendar.xml", $saveFileFormat->XML);
    	print "Created calendar, please check the output file.".PHP_EOL;
    }

}
?>